src/Microsoft.AspNet.Mvc.ModelBinding/Binders/TypeSpecificModelBinder.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+
+    public class TypeSpecificModelBinder : IModelBinder
+    {
+        public TypeSpecificModelBinder(Type modelType, IModelBinder modelBinder)
+        {
+            this.modelType = modelType;
+            this.modelBinder = modelBinder;
+        }
+
+        public Type modelType
+        {
+            get;
+            private set;
+        }
+
+        public IModelBinder modelBinder
+        {
+            get;
+            private set;
+        }
+
+        public async Task<bool> BindModelAsync(ModelBindingContext bindingContext)
+        {
+            if (bindingContext.ModelType != modelType)
```

Comment:
Exact match seems right for this. If you have a derived type then it probably has more details, and you don't have the necessary context to bind it.